src/main/java/net/dv8tion/jda/api/entities/GuildChannel.java

```diff
@@ -35,12 +36,23 @@ public interface GuildChannel extends Channel, Comparable<GuildChannel>
      * This getter is not thread-safe and would require guards by the user.
      *
      * @throws net.dv8tion.jda.api.exceptions.InsufficientPermissionException
-     *         If the currently logged in account does not have {@link net.dv8tion.jda.api.Permission#MANAGE_CHANNEL Permission.MANAGE_CHANNEL}
+     *         If the currently logged in account does not have the required {@link #getManagePermission() permission}
      *
      * @return The ChannelManager of this GuildChannel
      */
     @Nonnull
     ChannelManager<?, ?> getManager();
+    
+    /**
+     * Returns the {@link Permission} needed in order to manage or delete this channel.
+     * This returns {@link Permission#MANAGE_CHANNEL} or {@link Permission#MANAGE_THREADS} for {@link ThreadChannel ThreadChannels}
+     * 
+     * @return The {@link Permission}
+     */
+    @Nonnull
+    default Permission getManagePermission() {
```

Comment:
I'm against this design, I think instead we should pass the required permissions to the manager using `EnumSet<Permission>` as a constructor parameter.

Also, please try to follow the style of the project.

Reply:
Why use a `Set`? Do you think Discord will require multiple permissions for editing channels in the future?